presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -195,4 +246,76 @@ private void initializeColumnReaders()
             columnReadersMap.put(column, ParquetColumnReader.createReader(column));
         }
     }
+
+    private Block readBlock(String name, Type type, List<String> path, List<Integer> offsets, List<Integer> positions)
```

Comment:
If you use `it.unimi.dsi.fastutil.ints.IntList` with the `getInt` style methods, you will get better performance